sim/core/cast.go

```diff
@@ -196,16 +195,29 @@ func (spell *Spell) wrapCastFuncHaste(config CastConfig, onCastComplete CastFunc
 }
 
 func (spell *Spell) wrapCastFuncGCD(config CastConfig, onCastComplete CastFunc) CastFunc {
-	if config.DefaultCast.GCD == 0 {
+	if config.DefaultCast.empty() { // spells that are not actually cast (e.g. auto attacks, procs)
 		return onCastComplete
 	}
 
+	if config.DefaultCast.GCD == 0 { // mostly cooldowns (e.g. nature's swiftness, presence of mind)
+		return func(sim *Simulation, target *Unit) {
```

Comment:
This is going to add a check for every spell, including all the empty-cast spells like autoattacks. Is this a perf loss?

Reply:
Those are already handled by `DefaultCast.empty()` above - that's true for all autoattacks and procs, at least (which must not be checked here, since they have no such timing restrictions.